crates/starknet-server/tests/get_transaction_by_hash.rs

```diff
@@ -0,0 +1,102 @@
+pub mod common;
+
+mod get_transaction_by_hash_integration_tests {
+    use std::sync::Arc;
+
+    use starknet_core::constants::{DECLARE_V1_TRANSACTION_HASH, DECLARE_V2_TRANSACTION_HASH};
+    use starknet_rs_accounts::{Account, SingleOwnerAccount};
+    use starknet_rs_core::chain_id;
+    use starknet_rs_core::types::contract::{CompiledClass, SierraClass};
+    use starknet_rs_core::types::{
+        BlockId, BlockTag, BroadcastedDeclareTransactionV1, FieldElement,
+    };
+    use starknet_rs_providers::Provider;
+    use starknet_rs_signers::{LocalWallet, SigningKey};
+    use starknet_types::felt::Felt;
+    use starknet_types::traits::ToHexString;
+
+    use crate::common::constants::{
+        CASM_COMPILED_CLASS_HASH, PREDEPLOYED_ACCOUNT_ADDRESS, PREDEPLOYED_ACCOUNT_PRIVATE_KEY,
+    };
+    use crate::common::util::BackgroundDevnet;
+
+    #[tokio::test]
+    async fn get_declere_v1_transaction_by_hash_happy_path() {
+        let devnet = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let json_string = std::fs::read_to_string(concat!(
+            env!("CARGO_MANIFEST_DIR"),
+            "/test_data/rpc/declare_v1.json"
+        ))
+        .unwrap();
+        let declare_txn_v1: BroadcastedDeclareTransactionV1 =
+            serde_json::from_str(&json_string).unwrap();
+
+        let add_transaction = devnet
+            .json_rpc_client
+            .add_declare_transaction(starknet_rs_core::types::BroadcastedDeclareTransaction::V1(
+                declare_txn_v1.clone(),
+            ))
+            .await
+            .unwrap();
+
+        let get_transaction = devnet
+            .json_rpc_client
+            .get_transaction_by_hash(add_transaction.transaction_hash)
+            .await
+            .unwrap();
+
+        if let starknet_rs_core::types::Transaction::Declare(
+            starknet_rs_core::types::DeclareTransaction::V1(declare_v1),
+        ) = get_transaction
+        {
+            assert_eq!(
+                declare_v1.transaction_hash,
+                FieldElement::from_hex_be(DECLARE_V1_TRANSACTION_HASH).unwrap()
```

Comment:
Instead of asserting the tx hash, I would assert that the status is accepted.

Reply:
This would require a fix of declare v1 transaction introduced by Mario because now it's a reverted transaction.

Now I want to focus on DeployAccount and Invoke which are not implemented at all.

Later if it will be time for that, we can make improvements in tests: add more tests, different status checks etc. but in my opinion, is not a time for that now.

Reply:
Ok, no need to do status assertions here. We can leave that for when #92 is dealt with.